nntrainer/layers/cl_layers/custom_vocab_selection.cpp

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Yash Singh <yash.singh@samsung.com>>
+ *
+ * @file   custom_vocab_selection.cpp
+ * @date   1 Oct 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Yash Singh <yash.singh@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  Implementation of custom vocab selection
+ */
+
+#include <algorithm>
+#include <custom_vocab_selection.h>
+
+nntrainer::VocabSelection::VocabSelection(LshType lshType, int lshChoices,
+                                          int hiddenSize, int vocabCnt) :
+  lshType(lshType),
+  lshChoices(lshChoices),
+  vocabCnt(vocabCnt),
+  hiddenSize(hiddenSize),
+  lshBlockNum(0),
+  lshBits(0) {}
+
+nntrainer::VocabSelection::~VocabSelection() {}
+
+nntrainer::VocabSelectionNNTrainer::VocabSelectionNNTrainer(
+  LshType lshType, int lshChoices, int hiddenSize, int vocabCnt,
+  nntrainer::Tensor &weights) :
+  VocabSelection(lshType, lshChoices, hiddenSize, vocabCnt) {
+  this->lshBlockNum = (hiddenSize + lshBlockSize - 1) / lshBlockSize;
+  this->lshBits = lshBlockNum * lshBlockSize;
+  this->lshData = std::vector<lshDataBlock>(this->vocabCnt * lshBlockNum);
+
+  // for (unsigned int i = 0; i < vocabCnt; ++i) {
```

Comment:
If this comment is simply to record previous actions, would it not be better to remove it?

Reply:
Removed in the latest commit. Thanks for pointing it out.